File: LAB 7/Task 2 - Sense Specific Sensor/SenseMoreSensors/app/src/main/java/com/example/sensemoresensors/MainActivity.java
package com.example.sensemoresensors;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.os.Bundle;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements SensorEventListener {

    SensorManager senseMan;
    Sensor lightSensor;
    Sensor proximitySensor;
    Sensor temperatureSensor;
    TextView textProximity;
    TextView textPressure;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textProximity = (TextView) findViewById(R.id.proximity);
        textPressure = (TextView) findViewById(R.id.pressure);

        senseMan = (SensorManager) getSystemService(SENSOR_SERVICE);

        lightSensor = senseMan.getDefaultSensor(Sensor.TYPE_LIGHT);
        proximitySensor = senseMan.getDefaultSensor(Sensor.TYPE_PROXIMITY);
        temperatureSensor = senseMan.getDefaultSensor(Sensor.TYPE_PRESSURE);

        if (proximitySensor != null) {
            Toast.makeText(this, "Proximity Sensor Found", Toast.LENGTH_SHORT).show();
            senseMan.registerListener(this, proximitySensor, SensorManager.SENSOR_DELAY_NORMAL);
        } else {
            Toast.makeText(this, "Proximity Sensor Not Found", Toast.LENGTH_SHORT).show();
        }

        if (temperatureSensor != null) {
            Toast.makeText(this, "Temperature Sensor Found", Toast.LENGTH_SHORT).show();
            senseMan.registerListener(this, temperatureSensor, SensorManager.SENSOR_DELAY_NORMAL);
        } else {
            Toast.makeText(this, "Temperature Sensor Not Found", Toast.LENGTH_SHORT).show();
        }
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onSensorChanged(SensorEvent sensorEvent) {
        if (sensorEvent.sensor.getType() == Sensor.TYPE_PROXIMITY) {
            textProximity.setText("Proximity: " + Float.toString(sensorEvent.values[0]));
        } else if (sensorEvent.sensor.getType() == Sensor.TYPE_PRESSURE) {
            textPressure.setText("Pressure: " + Float.toString(sensorEvent.values[0]));
        }
    }


    @Override
    public void onAccuracyChanged(Sensor sensor, int i) {

    }

    @Override
    protected void onPause() {
        super.onPause();
        senseMan.unregisterListener(this);
    }

    @Override
    protected void onResume() {
        super.onResume();
        senseMan.registerListener(this, proximitySensor, SensorManager.SENSOR_DELAY_NORMAL);
        senseMan.registerListener(this, temperatureSensor, SensorManager.SENSOR_DELAY_NORMAL);
    }
}

File: LAB 7/Task 1 - Sensor List/SensorListKotlin/app/src/main/java/com/example/sensorlistkotlin/MainActivity.kt
package com.example.sensorlistkotlin

import android.hardware.Sensor
import android.hardware.SensorManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.ListView

class MainActivity : AppCompatActivity() {

    //SensorManager senseMan;
    private lateinit var senseMan: SensorManager

    //ListView lv;
    private lateinit var lv: ListView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //lv = (ListView) findViewById(R.id.listview);
        lv = findViewById(R.id.listview)

        //senseMan = (SensorManager) getSystemService(SENSOR_SERVICE);
        senseMan = getSystemService(SENSOR_SERVICE) as SensorManager

        //List<Sensor> sensorList = senseMan.getSensorList(Sensor.TYPE_ALL);
        val sensorList: List<Sensor> = senseMan.getSensorList(Sensor.TYPE_ALL)

        //lv.setAdapter(new ArrayAdapter<Sensor>(this, android.R.layout.simple_list_item_1, sensorList));
        lv.adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, sensorList)
    }
}